contracts/token/ERC721/extensions/ERC721WeightedVotes.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts (last updated v4.8.0) (token/ERC721/extensions/ERC721Votes.sol)
+
+pragma solidity ^0.8.0;
+
+import "../ERC721.sol";
+import "../../../governance/utils/Votes.sol";
+import "./IERC721Supply.sol";
+
+/**
+ * @dev Extension of ERC721 to support voting and delegation as implemented by {Votes}, where each individual NFT counts
+ * as 1 vote unit.
+ *
+ * Tokens do not count as votes until they are delegated, because votes must be tracked which incurs an additional cost
+ * on every transfer. Token holders can either delegate to a trusted representative who will decide how to make use of
+ * the votes in governance decisions, or they can delegate to themselves to be their own representative.
+ *
+ * _Available since v4.5._
+ */
+abstract contract ERC721WeightedVotes is ERC721, Votes, IERC721Supply {
```

Comment:
Requiring `IERC721Supply` means that this module either needs a `ERC721Enumerability`, or needs a lighter custom module (that we don't provide).

We are very strongly pushing toward NOT using `ERC721Enumerability`, so I don't think its right to require using it in modules like this.